### PR TITLE
Last easy mythos 

### PR DESCRIPTION
**Pull request by petercombs**

### CODE REVIEW

**Comment by petercombs** [comment]
This has some code copied from the 15-30 branch. Feel free to wait until that one is merged.

File: eldritch/mythos.py
Comment (petercombs) [inline eldritch/mythos.py:311]: This got past both of us before!

File: eldritch/eldritch.py
Comment (petercombs) [inline eldritch/eldritch.py:197]: Or I could put explicit can and cannot attributes (can_seal, can_use, cannot_trade_clues, maybe even can_heal_sanity and _stamina to allow e.g. the HealingStone to ask before using) either here or in BaseAncientOne.

File: eldritch/events.py
Comment (petercombs) [inline eldritch/events.py:4246]: I'm not sure what I was thinking here... is it redundant or just very careful having the check both here and at event creation?

File: eldritch/mythos.py
Comment (petercombs) [inline eldritch/mythos.py:130]: Should these be in events.py? They seem to be common, Mythos-specific patterns, but then so is ReleaseToStreets...

File: eldritch/mythos.py
Comment (petercombs) [inline eldritch/mythos.py:476]: Reminder to self: add to CreateMythos

**Review by petercombs — COMMENTED** [review]
Highlighted a few areas in particular.

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:627]: I think there probably needs to be a global modifier for invalid monsters (that cannot be drawn from the cup), and the DrawMonstersFromCup event just needs to check it.

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:688]: Don't you need to include the trigger that puts it back into the deck at the end of the next mythos?

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:756]: TODO: if a player use the red sign to ignore the undead attribute of a monster, does this effect no longer apply?

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:780]: Should this be state.turn_number instead of state.turn?

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:792]: DiscardSpecific takes a list of items, not a single item.

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:795]: When the second argument to PassFail is an event, PassFail automatically returns a sequence. You do not need to create your own.

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:797]: Also need to add the trigger to remove this from global effects when done.

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:890]: Want to add 64, 66, and 67 to this change?

**Review by fingoldfish — CHANGES_REQUESTED** [review]
More tests wouldn't hurt.

File: eldritch/mythos.py
Comment (petercombs) [reply eldritch/mythos.py:627]: It's not obvious to me that will do everything we need it to. In the FAQs, 

> Q: If investigators fail [Mythos 65 Rumor] while the [Mythos 44]  environment is in play, would that send [Flame Matrices] that were on the card to the cup?
> A: Yes.

Which implies that a Flame Matrix could also be placed onto Mythos65. 

Also, I think it's not obvious whether drawing a Flame Matrix while in an Other World would still mean you fight it.  Is it raining _so hard_ that it also affects things on Pluto?

Comment (petercombs) [reply eldritch/mythos.py]: I like this use of `can_*`.

File: eldritch/mythos.py
Comment (petercombs) [reply eldritch/mythos.py:965]: Now handled in the CloseLocationsHeadline class.  Fooled me once, shame on you, fooled me twice, it's time to automate that shit.

**Comment by petercombs** [comment]
Any other tests you want to see?

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:607]: Is this where you wanted this comment?

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:678]: No need to put both the check and the pass_fail in here - PassFail will automatically return a sequence if argument 2 is an event (as opposed to a value).

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:816]: You need a `return seq` here.

File: eldritch/mythos.py
Comment (fingoldfish) [inline eldritch/mythos.py:890]: should be `moon`, not crescent

File: eldritch/test_mythos.py
Comment (fingoldfish) [inline eldritch/test_mythos.py:1941]: Add one test that validates that if the board is full, the user gets a choice of monsters to place in the outskirts?

File: eldritch/test_mythos.py
Comment (fingoldfish) [inline eldritch/test_mythos.py:2185]: I feel like you meant to put this comment lower.

File: eldritch/test_mythos.py
Comment (fingoldfish) [inline eldritch/test_mythos.py:2219]: Does the character even have enough movement to make it to the shops?

File: eldritch/test_mythos.py
Comment (fingoldfish) [inline eldritch/test_mythos.py:2321]: Also assert that they actually get returned to the cup?

File: eldritch/test_mythos.py
Comment (fingoldfish) [inline eldritch/test_mythos.py:2440]: Once you've reached the SliderInput, do you need to test that there is nothing in state.usables?

File: eldritch/test_mythos.py
Comment (fingoldfish) [inline eldritch/test_mythos.py:2468]: I think you would need to use the vampire instead of the zombie here to determine how this is working. The red sign also reduces the monster's toughness by 1, so a zombie will have toughness 1 whether or not the red sign cancels the effect of the env.

File: eldritch/test_mythos.py
Comment (fingoldfish) [inline eldritch/test_mythos.py:2118]: I think we also want tests for mythos 16, 17, 35, 39, and 44.

**Review by fingoldfish — CHANGES_REQUESTED** [review]

File: eldritch/mythos.py
Comment (petercombs) [reply eldritch/mythos.py:607]: I guess here _and_ in 44 would make sense... it's not obvious that a heat wave or heavy rain in Massachussetts should affect what's happening in Antartica or Pluto.

File: eldritch/test_mythos.py
Comment (petercombs) [reply eldritch/test_mythos.py:2341]: They do now that I've given them a Ruby!

**Review by fingoldfish — CHANGES_REQUESTED** [review]
Just a few things left in the unit tests.

File: eldritch/test_mythos.py
Comment (petercombs) [reply eldritch/test_mythos.py:2360]: That's a good point... Though really the test should be on the CloseLocation event, not the Mythos test.  Going to update the test over in test_events and remove this test.

File: eldritch/test_mythos.py
Comment (petercombs) [reply eldritch/test_mythos.py:2581]: I guess I didn't think I needed to explicitly assert that usables is empty, but it doesn't hurt.  Also adding a DreadCurse to your inventory so the check in combat asserts that it's not usable there.

File: eldritch/test_mythos.py
Comment (petercombs) [reply eldritch/test_mythos.py:1635]: That's why I put Mythos44 in the left side of the deck, so it gets drawn. If you put an assertFalse(True) at the end, the event log is: 

```
E     Dummy's movement
E     Dummy's encounter phase
E       Dummy had an encounter at Diner
E         Dummy draws 1 encounter cards
E     Dummy's other world phase
E     Dummy's mythos phase
E       Dummy drew Mythos44
E       A gate appeared at Roadhouse.
E         Flame Matrix were drawn from the cup
E         1 doom tokens were added
E         Dummy sent 1 monsters to gates and 0 monsters to the outskirts.
E       A clue appeared at Square.
E       plus move on white; moon move on black
E       Mythos44 became the new environment
E       2 monsters returned to the cup
E     Dummy's upkeep
E       Dummy set sliders
E     Dummy's movement
E       Dummy lost 1 movement points
```

File: eldritch/test_mythos.py
Comment (petercombs) [reply eldritch/test_mythos.py:1600]: As with 44, it does get drawn from the left. 
```
E     Dummy's movement
E     Dummy's encounter phase
E       Dummy had an encounter at Diner
E         Dummy draws 1 encounter cards
E     Dummy's other world phase
E     Dummy's mythos phase
E       Dummy drew Mythos35
E       A gate appeared at Roadhouse.
E         Cultist were drawn from the cup
E         1 doom tokens were added
E         Dummy sent 1 monsters to gates and 0 monsters to the outskirts.
E       A clue appeared at Square.
E       moon move on white; plus move on black
E         Cultist moved from Roadhouse to Easttown
E       1 Elder Thing(s) respawned at Docks
E     Dummy's upkeep
E       Dummy set sliders
E     Dummy's movement
E     Dummy's encounter phase
E       Dummy had an encounter at Diner
E         Dummy draws 1 encounter cards
E     Dummy's other world phase
```

File: eldritch/test_mythos.py
Comment (fingoldfish) [reply eldritch/test_mythos.py:2578]: The reason to assert on usables being empty is that if the usables interrupt a choice event, they do not otherwise interrupt the flow. So if you resolve the choice event, you won't know whether anything was usable at the time of the choice.

File: eldritch/test_mythos.py
Comment (fingoldfish) [inline eldritch/test_mythos.py:2506]: Can I ask for a few more tests?
- mythos38 (clues to seal should decrease by 2)
- mythos57 (lose your whiskey when being arrested)
- mythos58 (health wager - test the reward and being devoured)

**Review by fingoldfish — CHANGES_REQUESTED** [review]

File: eldritch/test_mythos.py
Comment (petercombs) [reply eldritch/test_mythos.py:2071]: Annoyed grunt!  The `mock_ret` was getting used for both the gate open monster draw and the mythos monster draw.

File: eldritch/test_mythos.py
Comment (petercombs) [reply eldritch/test_mythos.py:2532]: Ask and ye shall receive.

File: eldritch/test_mythos.py
Comment (fingoldfish) [inline eldritch/test_mythos.py:2675]: Might make more sense to add this whiskey to the first test case, to make sure both whiskeys get discarded.

**Review by fingoldfish — APPROVED** [review]

**Pull request merged**
